### PR TITLE
Performance refactor of exporter

### PR DESCRIPTION
**Pull request by niwinz**

### CODE REVIEW

File: exporter/src/app/renderer/svg.cljs [hunk 1]
```diff
@@ -293,88 +294,74 @@
...
+
+          (render-object [page {:keys [id] :as object}]
+            (p/let [tmpdir (sh/mktmpdir! "svg-render")
+                    path   (path/join tmpdir (str/concat id (mime/get-extension type)))
+                    node   (bw/select page (str/concat "#screenshot-" id))]
+              (bw/wait-for node)
+              (p/let [xmldata (extract-svg page object)
+                      txtdata (extract-txt-nodes page object)
+                      result  (replace-text-nodes xmldata txtdata)
+                      result  (str/replace result "&nbsp;" "&#160;")]
```
Comment (hirunatan) [inline exporter/src/app/renderer/svg.cljs:337]: Seems that you have removed the comment explaining this (I think it's still necessary)

File: frontend/src/app/main/data/exports.cljs [hunk 2]
```diff
@@ -170,22 +160,26 @@
...
-                        (dom/trigger-download filename data)
-                        (clear-export-state uuid/zero)))
+         (->> (rp/query! :export-shapes params)
+              (rx/mapcat (fn [{:keys [id filename]}]
+                           (->> (rp/query! :exporter {:cmd :get-resource :blob? true :id id})
+                                (rx/map (fn [data]
+                                          (dom/trigger-download filename data)
+                                          (clear-export-state uuid/zero))))))
               (rx/catch (fn [cause]
-                          (prn "KKKK" cause)
```
Comment (hirunatan) [inline frontend/src/app/main/data/exports.cljs:181]: LOL (at last it's gone!)

File: frontend/src/app/util/dom.cljs [hunk 3]
```diff
@@ -464,11 +464,11 @@
...
 (defn trigger-download-uri
   [filename mtype uri]
-  (let [link (create-element "a")
+  (let [link      (create-element "a")
         extension (mtype->extension mtype)
-        filename (if extension
-                   (str filename "." extension)
-                   filename)]
+        filename  (if (and extension (not (str/ends-with? filename extension)))
+                    (str/concat filename "." extension)
```
Comment (hirunatan) [inline frontend/src/app/util/dom.cljs]: If the extension already has a `.`, no need to add it here again.

File: exporter/src/app/renderer/svg.cljs (hunk 1 above)
Comment (niwinz) [reply exporter/src/app/renderer/svg.cljs:337]: agree

File: frontend/src/app/util/dom.cljs (hunk 3 above)
Comment (niwinz) [reply frontend/src/app/util/dom.cljs]: fixed

**Pull request merged**